### PR TITLE
CSS convert 5 - dialog, drawer, dropdown, menu & dashify util

### PR DESCRIPTION
**Pull request by jaketrent**
asdf

### CODE REVIEW

File: packages/dropdown/src/react/selected.tsx [hunk 1]
```diff
@@ -1,59 +1,43 @@
...
-      appearance === vars.appearances.subtle
-        ? glamor.css(
-            stylesheet[
-              `.psds-dropdown__field--appearance-${vars.appearances.subtle}.psds-dropdown__placeholder--color`
-            ]
-          )
-        : glamor.css(stylesheet['.psds-dropdown__placeholder--color'])
-    return glamor.compose(
-      glamor.css(stylesheet[`.${label}`]),
-      isSmall && glamor.css(stylesheet[`.${label}.psds-dropdown--small`]),
```
Comment (EdwardIrby) [inline packages/dropdown/src/react/selected.tsx:30]: This is missing below

File: packages/dropdown/src/react/selected.tsx [hunk 2]
```diff
@@ -1,59 +1,43 @@
...
+      <span
+        {...rest}
+        aria-hidden
+        className={classNames('psds-dropdown__button-sizer', className)}
+      >
         {label || placeholder}
       </span>
-      <span {...styles.placeholder({ appearance, selectedItem, size })}>
+      <span
+        className={classNames(
```
Comment (EdwardIrby) [inline packages/dropdown/src/react/selected.tsx:33]: Missing psds-dropdown--small

File: packages/dropdown/src/react/sub-label.tsx [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
   const themeName = useTheme()
-  return subLabel ? <div {...styles(themeName)}>{subLabel}</div> : null
+  return subLabel ? (
+    <div
+      {...rest}
+      className={classNames(
+        'psds-dropdown__sub-label',
+        `psds-theme--${themeName}`,
+        className
+      )}
```
Comment (EdwardIrby) [inline packages/dropdown/src/react/sub-label.tsx:24]: While this one is fine I think inlining them like this makes it easier to miss stuff during the refactor like in some of my other comments.

File: packages/menu/src/react/check.tsx [hunk 4]
```diff
@@ -1,25 +1,21 @@
...
   return selected ? (
     // @ts-ignore: ignore icon type
-    <CheckIcon {...styles.icon} {...props} />
+    <CheckIcon
+      {...rest}
+      className={classNames('psds-menu__item-icon', className)}
+    />
   ) : (
-    <div {...styles.filler} style={props.style} />
+    <div className="psds-menu__item-icon-filler" style={rest.style} />
```
Comment (EdwardIrby) [inline packages/menu/src/react/check.tsx:19]: Should this also take className as it's a conditional return?

File: packages/navbar/src/react/__specs__/__snapshots__/storyshots.spec.tsx.snap [hunk 5]
```diff
@@ -441,7 +441,7 @@ exports[`Storyshots Components/NavBar Example Skills 1`] = `
         onClick={[Function]}
       >
         <div
-          className="psds-avatar psds-avatar--size-xsmall"
+          className="psds-avatar psds-avatar--size-x-small"
```
Comment (EdwardIrby) [inline packages/navbar/src/react/__specs__/__snapshots__/storyshots.spec.tsx.snap:444]: xsmall

File: packages/util/src/dashify.ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+export function dashify(str?: string) {
+  if (!str) return
+
+  return str
+    .trim()
+    .replace(/([a-z])([A-Z])/g, '$1-$2')
+    .replace(/([a-z])([0-9]+)/g, '$1-$2')
+    .replace(/([0-9])([a-z])/g, '$1-$2')
+    .toLowerCase()
+}
```
Comment (EdwardIrby) [inline packages/util/src/dashify.ts:10]: I think we should be doing this manually.

**Review by EdwardIrby — CHANGES_REQUESTED** [review]

File: packages/appframe/src/react/__specs__/__snapshots__/storyshots.spec.tsx.snap [hunk 7]
```diff
@@ -92,7 +92,7 @@ exports[`Storyshots Components/AppFrame Basic 1`] = `
               data-css-91smc8=""
             >
               <div
-                className="psds-avatar psds-avatar--size-xsmall"
+                className="psds-avatar psds-avatar--size-x-small"
```
[the remark this answers is not in the document]
Comment (jaketrent) [reply packages/appframe/src/react/__specs__/__snapshots__/storyshots.spec.tsx.snap:95]: Yes, `xsmall` is right. I'll update the util. Thanks.

File: packages/util/src/dashify.ts (hunk 6 above)
Comment (jaketrent) [reply packages/util/src/dashify.ts:10]: I don't know exactly what you mean by your comment. I think I need more context.

File: packages/dropdown/src/react/sub-label.tsx (hunk 3 above)
Comment (jaketrent) [reply packages/dropdown/src/react/sub-label.tsx:24]: I'm a little unclear on your meaning. You're saying inlining `className` values makes it easier to miss stuff. What stuff? You refer to other comments. Which ones? If I knew which ones, maybe I'd see what you meant by missing stuff. Perhaps you're talking about this one? https://github.com/pluralsight/design-system/pull/1881#discussion_r662915914 If so, what are you thoughts on how not inlining the className value would help not miss a selector?

File: packages/dialog/src/react/index.tsx [hunk 8]
```diff
@@ -135,8 +106,14 @@ const Dialog = React.forwardRef((props, ref) => {
 
   const content = (
     <FocusManager
-      {...styles.dialog(modal, tailPosition)}
       {...rest}
+      className={classNames(
+        'psds-dialog',
+        modal && 'psds-dialog--modal',
+        typeof tailPosition !== 'undefined' &&
+          `psds-dialog--w-tail psds-dialog--tailposition-${tailPosition.toLowerCase()}`,
```
[the remark this answers is not in the document]
Comment (jaketrent) [reply packages/dialog/src/react/index.tsx:114]: Right, will update to dashify.

**Review by sweeting — CHANGES_REQUESTED** [review]
Did a visual check of dialog, drawer, menu. 

menu: the menu items with a description have 32px padding on left/right instead of 16px like the other menu items.

menu padding

File: packages/drawer/src/react/index.tsx [hunk 9]
```diff
@@ -58,21 +38,32 @@ export const useDrawerContext = () => {
...
-  ({ children, ...rest }, ref) => {
+  ({ children, className, ...rest }, ref) => {
     const themeName = useTheme()
     const { isOpen, onToggle } = useDrawerContext()
     return (
       <div
-        {...styles.head(themeName, isOpen)}
+        className={classNames(
+          `psds-drawer__summary`,
+          isOpen && 'psds-drawer__summary.psds-drawer--isopen',
```
[the remark this answers is not in the document]
Comment (jaketrent) [reply packages/drawer/src/react/index.tsx:48]: Great catch. Yes, will fix.

File: packages/dropdown/src/css/index.css [hunk 10]
```diff
@@ -0,0 +1,234 @@
...
+  font-weight: var(--ps-type-font-weight-default);
+  color: var(--ps-colors-text-icon-high-on-light);
+  text-align: left;
+  cursor: pointer;
+  padding: 0 var(--ps-layout-spacing-small);
+}
+.psds-dropdown__field:focus {
+  outline: 0;
+}
+.psds-dropdown--small .psds-dropdown__field {
```
[the remark this answers is not in the document]
Comment (jaketrent) [reply packages/dropdown/src/css/index.css:33]: No, I changed the implementation more. The parent node is now given the size:

https://github.com/pluralsight/design-system/pull/1881/files#diff-cee44dcd0c7455f8a9f6d9a46a4b413f94bb927319d40d6476379c45c57e175aR84
https://github.com/pluralsight/design-system/pull/1881/files#diff-5ad9823e84e926f4fa97c3b55a07080f7d4348b334b6744e28a287072330eafdR36

This is made possible by loosing the constraints of glamor. I'm not 100% sure it's better for selector maintainability. But it does take advantage of CSS more fully and it simplifies the react side a bit.

File: packages/dropdown/src/react/selected.tsx (hunk 1 above)
Comment (jaketrent) [reply packages/dropdown/src/react/selected.tsx:30]: Non-issue. See https://github.com/pluralsight/design-system/pull/1881#discussion_r664665330

File: packages/dropdown/src/react/selected.tsx (hunk 2 above)
Comment (jaketrent) [reply packages/dropdown/src/react/selected.tsx:33]: Non-issue. See https://github.com/pluralsight/design-system/pull/1881#discussion_r664665330

File: packages/menu/src/react/check.tsx (hunk 4 above)
Comment (jaketrent) [reply packages/menu/src/react/check.tsx:19]: Yes, I think so. Good catch.

**Comment by jaketrent** [comment]
I believe I have adjusted everything for which changes were previously requested. Please have another look.

File: packages/dropdown/src/css/index.css (hunk 10 above)
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/dropdown/src/css/index.css:33]: True. I worry about using this pattern when it's not a direct child. But let's go with it.

**Comment by EdwardIrby** [comment]
@jaketrent still seeing this in storybook pretty sure placeholder styles are off. @sweeting you seeing this too?
Screen Shot 2021-07-06 at 3 15 29 PM

**Comment by jaketrent** [comment]
@EdwardIrby yes, I am seeing this viz problem on Dropdown. Thanks for the catch. I should have seen this.

**Comment by jaketrent** [comment]
Hey @EdwardIrby, I pushed a fix for that bg problem on the subtle dropdown.

**Review by EdwardIrby — APPROVED** [review]
Let's roll!

**Review by sweeting — CHANGES_REQUESTED** [review]
**Menu**
The <button> has additional padding that seems to be from default browser styles that aren't reset. In Firefox it's 1px 4px. In Safari, it's 2px 6px
CleanShot 2021-07-07 at 09 27 50

I didn't catch this earlier, but there is only a hover style on the menu item, but no focus style. The focus and hover styles should be the same. 

The arrow keys can't be used to navigate the menu items. 
https://www.w3.org/TR/wai-aria-practices-1.2/#menu

**Comment by jaketrent** [comment]
Hey @sweeting , thanks for the sharp eye on those things.

Padding has been normalized to 0. Focus style has been created. May have never worked.

I carded the keyboard nav for a separate issue in #1900.

Please have another look.

**Review by sweeting — APPROVED** [review]
👍

**Pull request merged**